P/polymake/build_tarballs.jl

```diff
@@ -94,9 +94,10 @@ install_license COPYING
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = [
- MacOS(:x86_64),
- Linux(:x86_64, libc=:glibc, compiler_abi=CompilerABI(cxxstring_abi=:cxx11)),
+    Platform("x86_64", "linux"; libc="gnu"),
+    Platform("x86_64", "macos"),
 ]
+platforms = expand_cxxstring_abis(platforms)
```

Comment:
@giordano thanks for helping out!

@benlorenz was there a particular reason to only allow the cxx11 strings ABI (note: as far as I understand things, the cxxstring ABI does *not* restrict the C++ standard one can use; i.e. one can use the cxx03 string ABI also with C++11 or C++14 code, and vice versa use the cxx11 string ABI in C++03 code.

Reply:
Indeed, see the blue box under https://juliapackaging.github.io/BinaryBuilder.jl/dev/building/#Expanding-C-string-ABIs-or-libgfortran-versions

Reply:
There is no `libcxxwrap_julia` for the cxx03 ABI: https://github.com/JuliaPackaging/Yggdrasil/blob/4f8c5dba511db6bf93cff312b4b6268e16d94df4/L/libcxxwrap_julia/build_tarballs.jl#L36

Reply:
That may hopefully change in the future

Reply:
Note that a bunch of other packages are using `CompilerABI`, e.g. `CMake`, `libcxxwrap_julia`, `libpolymake_julia`, `libsingular_julia`, and more, so I guess all of those should be adjusted? Of course it's a bit annoying that this then means that people who want to test them locally now have to install dev versions of Julia 1.6, BinaryBuilder, and BinaryBuilderBase...

Reply:
> Of course it's a bit annoying that this then means that people who want to test them locally now have to install dev versions of Julia 1.6, BinaryBuilder, and BinaryBuilderBase...

I know, but unfortunately in BinaryBuilder now we _heavily_ rely on features present in Julia nightly which makes keeping compatibility with older Julia versions nearly impossible, or too much effort to be worth it.  The compatibility shim now used for the platforms doesn't really work great, I'm not going to spend time on it (my bandwidth is limited...), but anyone is welcome to improve it .

Note that to test one of these builders locally the quick and dirty trick is to locally set
```
platforms = Platform[]
```
and manually build from the command line for whatever platform you're interested to.